pkg/cluster/cluster.go

```diff
@@ -90,6 +91,11 @@ type manager struct {
 	securitycli    securityclient.Interface
 	arocli         aroclient.Interface
 	registryclient imageregistryclient.Interface
+	dynamichelper  dynamichelper.Interface
+
+	// We pass in these clients when we do testing, so add an option to omit
+	// creating them
+	shouldInitKubernetesClients bool
```

Comment:
Something is wrong if we have to do it, but I haven't figured out what exactly yet. Still a lot to review.